Add content: How To Search with Sourcegraph using Structural Patterns

marekweb opened the pull request:
## Type of change

Add new content

### What should this PR do?

Add a post titled How To Search with Sourcegraph using Structural Patterns.

### What are the acceptance criteria? 

- [ ] The new page displays correctly
- [ ] The card displays correctly in card lists
- [ ] All links work
- [ ] The front-matter fields are accurate and complete
- [ ] The page shows correctly on social media previews (checked using a validator)

### How should this PR be tested?

- Open the deploy preview and browse.

## Pull request process

**Reviewers**:

1. Test functionality using the criteria above.
2. Offer tips for efficiency, feedback on best practices, and possible alternative approaches and things that may not have been considered.
3. For shorter, "quick" PRs, use your best judgement on #​2.
4. Use a collaborative approach and provide resources and/or context where appropriate.
5. Provide screenshots/grabs where appropriate to show findings during review.

**Reviewees**:

1. Prefer incremental and appropriately-scoped changes.
2. Leave a comment on things you want explicit feedback on.
3. Respond clearly to comments and questions.

marekweb commented:
I noticed some things that could be improved in terms of visual appearance -- I'll mention them here although they're outside of the scope of this PR.

- The inline icon that I added to represent the structural search button at the beginning of this article (it uses the same icon component as actual button uses in the Sourcegraph app) is a bit too high in terms of vertical alignment. We could solve this in CSS. But we could also solve it by removing this icon entirely, because maybe it's not worth tweaking and maintaining the CSS for a small detail like this -- the icon might be redundant anyway since there's an animation of the button that follows it.
- The search boxes could benefit from having larger inputs. In this article in particular which has longer search examples, the input boxes are too small to fit the examples without having to scroll with the cursor. The inner padding in the inputs is reducing the amount of available space further. Maybe reworking the search box component would help, and I wonder if it would also help to make it a multi-line input.

ltagliaferri commented:
> I noticed some things that could be improved in terms of visual appearance -- I'll mention them here although they're outside of the scope of this PR.
> 
>     * The inline icon that I added to represent the structural search button at the beginning of this article (it uses the same icon component as the actual button uses in the Sourcegraph app) is a bit too high in terms of vertical alignment. We could solve this in CSS. But we could also solve it by removing this icon entirely, because maybe it's not worth tweaking and maintaining the CSS for a small detail like this -- the icon might be redundant anyway since there's an animation of the button that follows it.
> 
>     * The search boxes could benefit from having larger inputs. In this article in particular which has longer search examples, the input boxes are too small to fit the examples without having to scroll with the cursor. The inner padding in the inputs (maybe a bug?) is reducing the amount of available space further. Maybe redesigning the search box component would help, and I wonder if it would also help to make it a multi-line input.

cc @katjuell ^^

ltagliaferri commented on posts/structural-search-patterns.md:
```diff
@@ -0,0 +1,106 @@
+
+## Enabling structural search on Sourcegraph
+
+Enable structural search by clicking the icon of two square brackets (<CodeBracketsIcon/>) to the right of the search box:
```
Can we do the `<CodeBracketsIcon/>` as markdown backtics instead?

ltagliaferri commented on posts/structural-search-patterns.md:
```diff
@@ -0,0 +1,106 @@
+description: Learn how to use structural search patterns to search code on Sourcegraph.
+image: /headers/sourcegraph-learn-header-8.svg
+imageAlt: Sourcegraph Learn
+socialImage: /headers/sourcegraph-learn-header-8.png
```
I would recommend header 10 because it's blue, otherwise we'll have two yellow ones in this series, and they sit next to each other on the homepage too.

ltagliaferri commented on posts/structural-search-patterns.md:
```diff
@@ -0,0 +1,106 @@
+
+Enable structural search by clicking the icon of two square brackets (<CodeBracketsIcon/>) to the right of the search box:
+
+<GifLikeVideo url="/tutorial-images/enable-structural-search.mp4"/>
```
This takes a while to understand that this is an animation, maybe it could be clipped at the beginning

ltagliaferri replied:
I had to sit on the page for a bit to realize that this is animated (I knew based on it being an mp4), would it be possible to clip the video at the beginning so it is more obvious when someone is scrolling down?

ltagliaferri reviewed (changes requested):
I think we should not commit both code and content changes in the same PR at this point. @katjuell can you review the changes to `index.tsx`?

Also see my suggestions on the contnent.

marekweb commented:
To clarify, there's no code change here, it's just the addition of CodeBracketsIcon.

If there's no objection it sounds like we are OK to just get rid of the icon anyway, so I'll remove it, and for consistency I can remove the regex icon from the regex tutorial, too.

marekweb commented:
I updated the branch:

- Changed to header 10 (blue).
- Removed the use of the icon for structural search and did the same for the regex article, replacing them with `[]` and `.*` respectively.
- Trimmed the animation so that it starts right away.

ltagliaferri commented:
> To clarify, there's no code change here, it's just the addition of CodeBracketsIcon.
> 
> If there's no objection it sounds like we are OK to just get rid of the icon anyway, so I'll remove it, and for consistency I can remove the regex icon from the regex tutorial, too.

Yeah, I just meant working within `.ts` files and adding dependencies I think is out of scope for a content-only PR. I know we are using the `mdi-react` library but am not sure if adding in the brackets that way is adding to it. I think inline as markdown looks better as it's not floating, and doesn't require an extra `import` statement
